tests/generate_packageTest.py

```diff
@@ -0,0 +1,99 @@
+import os
+import shutil
+import sys
+import unittest
+from unittest.mock import patch
+from bears.generate_package import (touch, create_file_from_template,
+                                    create_file_structure_for_packages,
+                                    perform_upload, main,
+                                    create_upload_parser)
+
+
+class touchTest(unittest.TestCase):
+
+    def test_file_doesnt_exist(self):
+        if os.path.exists('TestFile.py'):
+            os.remove('TestFile.py')
+        self.assertFalse(os.path.exists('TestFile.py'))
+        touch('TestFile.py')
+        self.assertTrue(os.path.exists('TestFile.py'))
+
+    def tearDown(self):
+        os.remove('TestFile.py')
+
+
+class create_file_from_templateTest(unittest.TestCase):
+
+    def test_output_file(self):
+        data = {'who': 'George', 'sport': 'swimming'}
+        create_file_from_template(
+            os.path.join(
+                'tests', 'generate_package_input_files', 'template_file.py.in'),
+            os.path.join(
+                'tests', 'generate_package_input_files', 'substituted_file.py'),
+            data)
+        with open(
```

Comment:
Maybe just my style preference: if you have a four line `with` statement, you might want to put the path in `open(...)` in an extra variable.